source/OptimaJet.DWKit.StarterApplication/Forms/Projects/UpdateForm.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Linq;
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal;
+using Microsoft.EntityFrameworkCore;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.DWKit.StarterApplication.Services;
+using Serilog;
+
+namespace OptimaJet.DWKit.StarterApplication.Forms.Projects
+{
+    public class UpdateForm : BaseForm
+    {
+        public UserRepository UserRepository { get; set; }
+        public GroupRepository GroupRepository { get; set; }
+        public IEntityRepository<Organization> OrganizationRepository { get; set; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public ProjectRepository ProjectRepository { get; set; }
+        public UpdateForm(
+            UserRepository userRepository,
+            GroupRepository groupRepository,
+            ICurrentUserContext currentUserContext,
+            IEntityRepository<Organization> organizationRepository,
+            ProjectRepository projectRepository)
+        {
+            UserRepository = userRepository;
+            GroupRepository = groupRepository;
+            CurrentUserContext = currentUserContext;
+            OrganizationRepository = organizationRepository;
+            ProjectRepository = projectRepository;
+        }
+
+        public bool IsValid(int id, Project project)
+        {
+            var retVal = true;
+            if ((project.OrganizationId != 0)
+                || (project.OwnerId != 0)
+                || (project.GroupId != 0))
+            {
+                var original = ProjectRepository.Get()
+                    .Where(p => p.Id == id)
+                    .Include(p => p.Organization)
+                    .Include(p => p.Owner)
+                    .Include(p => p.Group)
+                       .ThenInclude(g => g.Owner)
+                    .FirstOrDefaultAsync().Result;
+                var owner = UserRepository.Get()
```

Comment:
you should already have access to the owner from `original.Owner`? why make this extra query?

Reply:
Rewrote the query so that it was possible to get all of the group membership and organization membership includes in one query